lib/assets/javascripts/turbograft/page.coffee

```diff
@@ -25,3 +25,42 @@ Page.refresh = (options = {}, callback) ->
 
 Page.open = ->
   window.open(arguments...)
+
+# Providing hooks for objects to set up destructors:
+onReplaceCallbacks = []
+
+# e.g., Page.onReplace(node, unbindListenersFnc)
+# unbindListenersFnc will be called if the node in question is partially replaced
+# or if a full replace occurs.  It will be called only once
+Page.onReplace = (node, callback) ->
+  throw new Error("Node and callback must both be specified") if !node || !callback
```

Comment:
`typeof func == 'function'` is probably fine
We could go for the hyper performant version but it's not as clear and probably not necessary for this. I don't imagine this would be used thousands of times per second, to the point it was affecting performance, and if it was I would bet the partial replacement node check would have even worse performance.